### PR TITLE
Bump Ubuntu from 18.04 to 22.04 and Azure CLI from 2.0.59 to 2.32.0

### PR DESCRIPTION
**Pull request by basil**
There is no specific driver for this PR, but I generally prefer having the latest toolchain when working. In my experience this typically makes development significantly easier compared to working on outdated toolchains. While Ubuntu 22.04 has not officially been released yet, we're close enough to April 2022 and far enough away from April 2018 that I think this is OK.

Upgrading the Ubuntu distribution also required updating the Azure CLI to the latest version. Since I couldn't find any Microosft repositories for `jammy` or `impish`, I did the next-best thing and picked the repository for `hirsute`. Once Microsoft provides a repository for `jammy`, we can switch to that in the future.

To test this change, I successfully built the `.deb`, `.rpm`, and SUSE packages. I also ran the Molecule tests against the resulting packages on Ubuntu 22.04, Fedora 35, and openSUSE Leap 15. All testing was successful.

CC @dduportal

### CODE REVIEW

File: Dockerfile
Comment (timja) [inline Dockerfile:20]: Doesn’t this create a separate command than the old package?
Do the tests cover creating a repo?

Comment (basil) [reply Dockerfile:23]: Oh good point. I added a symlink as a compatibility shim to help us transition. Once this image is adopted, callers can be adjusted to call `createrepo_c`, after which the compatibility shim can be removed.

**Comment by timja** [comment]
Build is failing:
```

#7 [stage-2  2/16] RUN apt-get update   && apt-get install --yes --no-insta...

#7 1.529 Get:1 http://security.ubuntu.com/ubuntu jammy-security InRelease [90.7 kB]

#7 1.573 Get:2 http://archive.ubuntu.com/ubuntu jammy InRelease [270 kB]

#7 1.918 Err:1 http://security.ubuntu.com/ubuntu jammy-security InRelease

#7 1.918   The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C

#7 1.984 Get:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [90.7 kB]

#7 2.080 Get:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [90.7 kB]

#7 2.279 Err:2 http://archive.ubuntu.com/ubuntu jammy InRelease

#7 2.279   The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C

#7 2.519 Err:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease

#7 2.519   The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C

#7 2.823 Err:4 http://archive.ubuntu.com/ubuntu jammy-backports InRelease

#7 2.823   The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C

#7 2.874 Reading package lists...

#7 2.893 W: http://security.ubuntu.com/ubuntu/dists/jammy-security/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-cdimage.gpg are ignored as the file is not readable by user '_apt' executing apt-key.

#7 2.893 W: http://security.ubuntu.com/ubuntu/dists/jammy-security/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2018-archive.gpg are ignored as the file is not readable by user '_apt' executing apt-key.

#7 2.894 W: GPG error: http://security.ubuntu.com/ubuntu jammy-security InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C

#7 2.894 E: The repository 'http://security.ubuntu.com/ubuntu jammy-security InRelease' is not signed.

#7 2.894 W: http://archive.ubuntu.com/ubuntu/dists/jammy/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-cdimage.gpg are ignored as the file is not readable by user '_apt' executing apt-key.

#7 2.894 W: http://archive.ubuntu.com/ubuntu/dists/jammy/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2018-archive.gpg are ignored as the file is not readable by user '_apt' executing apt-key.

#7 2.894 W: GPG error: http://archive.ubuntu.com/ubuntu jammy InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C

#7 2.894 E: The repository 'http://archive.ubuntu.com/ubuntu jammy InRelease' is not signed.

#7 2.894 W: http://archive.ubuntu.com/ubuntu/dists/jammy-updates/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-cdimage.gpg are ignored as the file is not readable by user '_apt' executing apt-key.

#7 2.894 W: http://archive.ubuntu.com/ubuntu/dists/jammy-updates/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2018-archive.gpg are ignored as the file is not readable by user '_apt' executing apt-key.

#7 2.894 W: GPG error: http://archive.ubuntu.com/ubuntu jammy-updates InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C

#7 2.894 E: The repository 'http://archive.ubuntu.com/ubuntu jammy-updates InRelease' is not signed.

#7 2.894 W: http://archive.ubuntu.com/ubuntu/dists/jammy-backports/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2012-cdimage.gpg are ignored as the file is not readable by user '_apt' executing apt-key.

#7 2.894 W: http://archive.ubuntu.com/ubuntu/dists/jammy-backports/InRelease: The key(s) in the keyring /etc/apt/trusted.gpg.d/ubuntu-keyring-2018-archive.gpg are ignored as the file is not readable by user '_apt' executing apt-key.

#7 2.894 W: GPG error: http://archive.ubuntu.com/ubuntu jammy-backports InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 871920D1991BC93C

#7 2.894 E: The repository 'http://archive.ubuntu.com/ubuntu jammy-backports InRelease' is not signed.

#7 ERROR: executor failed running [/bin/bash -eo pipefail -c apt-get update   && apt-get install --yes --no-install-recommends     apt-utils     createrepo-c     curl     build-essential     debhelper     devscripts     expect     fakeroot     git     gpg     gpg-agent     make     openssh-server     openssl     python3-pip     python3-pytest     python3-venv     rpm     rsync     tzdata     unzip   && ln -s /usr/bin/createrepo_c /usr/bin/createrepo   && apt-get clean   && pip3 install --no-cache-dir jinja2   && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*]: runc did not terminate successfully



#17 [jenkins-agent 1/1] FROM docker.io/jenkins/inbound-agent:4.11.2-2-jdk11@...

#17 unpacking docker.io/jenkins/inbound-agent:4.11.2-2-jdk11@sha256:2567dbbcf76bcaf7d9178ecd74d8e4d44868fc373a43e5ad8dbd40e8e884f4c1 3.2s done

#17 CANCELED

------

 > [stage-2  2/16] RUN apt-get update   && apt-get install --yes --no-install-recommends     apt-utils     createrepo-c     curl     build-essential     debhelper     devscripts     expect     fakeroot     git     gpg     gpg-agent     make     openssh-server     openssl     python3-pip     python3-pytest     python3-venv     rpm     rsync     tzdata     unzip   && ln -s /usr/bin/createrepo_c /usr/bin/createrepo   && apt-get clean   && pip3 install --no-cache-dir jinja2   && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*:

------

Error: failed to solve: executor failed running [/bin/bash -eo pipefail -c apt-get update   && apt-get install --yes --no-install-recommends     apt-utils     createrepo-c     curl     build-essential     debhelper     devscripts     expect     fakeroot     git     gpg     gpg-agent     make     openssh-server     openssl     python3-pip     python3-pytest     python3-venv     rpm     rsync     tzdata     unzip   && ln -s /usr/bin/createrepo_c /usr/bin/createrepo   && apt-get clean   && pip3 install --no-cache-dir jinja2   && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*]: runc did not terminate successfully

make: *** [Makefile:34: build] Error 1

script returned exit code 2
```

**Comment by basil** [comment]
> Build is failing

Works fine for me locally. I do not have permissions to re-run the build in CI.

**Comment by timja** [comment]
> > Build is failing
> 
> Works fine for me locally. I do not have permissions to re-run the build in CI.

It's failed twice, I've triggered it again

**Comment by dduportal** [comment]
Given the low bandwidth of the infra-team until the 6th, do you mind if we hold this change?

For info, there are multiple risks (in term of time to spend, not techncially!) involved that would problematic to prioritize over other infra tasks for these upcoming 10 days:

- Access to infra.ci.jenkins.io: this instance is aimed at being "not available" except for a few contributors. We currently have Stephane in the team whome don't have all accesses, and now @basil you are a (fine!) candidate for accessing this instance.

- The build error seems to appear with the Docker-less solution (`img`) that is currently used on infra.ci (ref. https://github.com/jenkins-infra/pipeline-library/blob/master/vars/buildDockerAndPublishImage.groovy). It is planned for februrary that the shared library moves to `docker` so it will use a valid Docker CE engine on virtual machines (because Docker in a Kubernetes Pod is not a valid solution due to its unsecure settings).

- Once the `docker` label can be used, there would be no reason to NOT let ci.jenkins.io to lint/build/test while infra.ci deploys (same model as jenkins plugin with ci.j / trusted.j) . That would make the contributors much more prone to restart failed build or access logs

**Comment by basil** [comment]
I just rebuilt this change again locally with `docker build .--no-cache`, so I think the change is correct. If the change is not correct due to a locally reproducible problem, please let me know how to reproduce it and I will correct the change. If the change is correct but can't be built on CI due to some operational issue, please resolve the operational issue at your convenience (no rush) and then merge this PR once the operational issue has been resolved.

**Comment by timja** [comment]
It’s not using docker build but ‘img build’

https://github.com/genuinetools/img

**Comment by dduportal** [comment]
> I just rebuilt this change again locally with `docker build .--no-cache`, so I think the change is correct. If the change is not correct due to a locally reproducible problem, please let me know how to reproduce it and I will correct the change. If the change is correct but can't be built on CI due to some operational issue, please resolve the operational issue at your convenience (no rush) and then merge this PR once the operational issue has been resolved.

+1 . The error seems related to `img` + our cluster (can't reproduce locally either on Docker or on `img`). Let's see in 2 weeks.

Thanks for the change @basil !

**Comment by dduportal** [comment]
For info, https://github.com/jenkins-infra/docker-packaging/pull/32 should help in fixing this behavior.
Once approved and merged, we'll be able to rebase / update the PR here and that should work as expected!

**Review by dduportal — CHANGES_REQUESTED** [review]
Microsoft is now supporting Ubuntu 22.04 as per https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-linux?pivots=apt 
(Good news: we passed the release date of this Ubuntu version, bad news: it means that we've let your PR here rot since then: sorry for that).

=> that should help removing some code (which is breaking the build as for now).

**Comment by dduportal** [comment]
Side not: I've invited @basil to this repository as writer . Once accepted, we'll have to trigger the mulitbranch scan on infra.ci 1 time so he's recognized as "trusted" collaborator for this repo (and his changes will be picked to trigger builds)

**Pull request closed without merging**
